src/Collections/Sequences/Seq.dfy

```diff
@@ -80,298 +86,310 @@ module {:options "-functionSyntax:4"} Seq {
   *
   ***********************************************************/
   
-  ghost predicate IsPrefix<T>(a: seq<T>, b: seq<T>)
-    ensures IsPrefix(a, b) ==> (|a| <= |b| && a == b[..|a|])
+  /* Is true if the sequence xs is a prefix of the sequence ys. */
+  ghost predicate IsPrefix<T>(xs: seq<T>, ys: seq<T>)
+    ensures IsPrefix(xs, ys) ==> (|xs| <= |ys| && xs == ys[..|xs|])
   {
-    a <= b    
+    xs <= ys    
   }
   
-  ghost predicate IsSuffix<T>(a: seq<T>, b: seq<T>)
+  /* Is true if the sequence xs is a suffix of the sequence ys. */
+  ghost predicate IsSuffix<T>(xs: seq<T>, ys: seq<T>)
   {
-    && |a| <= |b|
-    && a == b[|b|-|a|..]
+    && |xs| <= |ys|
+    && xs == ys[|ys|-|xs|..]
   }
 
-  /* a sequence that is sliced at the jth element concatenated with that same
-  sequence sliced from the jth element is equal to the original, unsliced sequence */
-  lemma LemmaSplitAt<T>(s: seq<T>, pos: nat)
-    requires pos < |s|;
-    ensures s[..pos] + s[pos..] == s;
+  /* A sequence that is sliced at the pos-th element, concatenated 
+     with that same sequence sliced from the pos-th element, is equal to the 
+     original unsliced sequence. */
+  lemma LemmaSplitAt<T>(xs: seq<T>, pos: nat)
+    requires pos < |xs|;
+    ensures xs[..pos] + xs[pos..] == xs;
   {
   }
 
-  /* ensures that the element from a slice is included in the original sequence */
-  lemma LemmaElementFromSlice<T>(s: seq<T>, s':seq<T>, a: int, b: int, pos: nat)
-    requires 0 <= a <= b <= |s|;
-    requires s' == s[a..b];
+  /* Any element in a slice is included in the original sequence. */
```

Comment:
I would expect this lemma to prove `ys[pos-a] in xs`
Even better would be that pos is an index into ys rather than xs, but that would change the API.
also keeping xs' instead of ys connotes that xs' is derived from xs